felix/dataplane/linux/int_dataplane.go

```diff
@@ -977,6 +979,30 @@ func ConfigureDefaultMTUs(hostMTU int, c *Config) {
 	}
 }
 
+func cleanUpIPIPAddrs() {
+	// If IPIP is not enabled, check to see if there is a VXLAN device and delete it if there is.
+	log.Debug("Checking if we need to clean up the IPIP device")
+	link, err := netlink.LinkByName("tunl0")
+	if err != nil {
+		if _, ok := err.(netlink.LinkNotFoundError); ok {
+			log.Debug("IPIP disabled and no IPIP device found")
+			return
+		}
+		log.WithError(err).Warn("IPIP disabled and failed to query IPIP device. Ignoring.")
+		return
+	}
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		log.WithError(err).Warn("IPIP disabled and failed to list addresses. Ignoring.")
```

Comment:
Might be worth including the consequence of this log, so users understand it when they see it.

e.g., "failed to list addresses, will be unable to remove any old addresses from the device should they exist"

Reply:
Will do, thanks!